### PR TITLE
Qualified name and flexible type rendering for java types.

### PR DESCRIPTION
**Pull request by neetopia**

### CODE REVIEW

File: test-utils/testData/api/resolveJavaType.kt [hunk 1]
```diff
@@ -60,12 +60,6 @@
 // END
 // FILE: a.kt
 annotation class Test
-@Test
```
Comment (neetopia) [inline test-utils/testData/api/resolveJavaType.kt:63]: This part seems to be unrelated to the test as removing them does not cause issues. In fact, putting them here actually causes issues in analysis API test because these declarations are a duplicating declaration with the actual classes we want to test in Java sources, FE1.0 was not seeing this issue because it looks legit from PSI manager (duplicate declarations are from Kotlin).

**Review by ting-yuan — APPROVED** [review]
